### PR TITLE
Silence the libevent warning when a proc abnormally terminates by cleaning up the error registrations

### PR DESCRIPTION
**Pull request by rhc54**

### CODE REVIEW

File: opal/mca/pmix/pmix112/pmix/src/usock/usock_sendrecv.c [hunk 1]
```diff
@@ -99,15 +101,22 @@ static void lost_connection(pmix_peer_t *peer, pmix_status_t err)
...
+        pmix_list_remove_item(&(peer->info->nptr->server->ranks), &(peer->info->super));
+        PMIX_RELEASE(peer->info);
+        /* reduce the number of local procs */
+        --peer->info->nptr->server->nlocalprocs;
+        /* do some cleanup as the client has left us */
+        pmix_pointer_array_set_item(&pmix_server_globals.clients,
                                      peer->index, NULL);
-         PMIX_RELEASE(peer);
+        /* remove all registered event handlers so libevent doesn't complain */
+        PMIX_LIST_FOREACH_SAFE(reginfoptr, regnext, &pmix_server_globals.client_eventregs, pmix_regevents_info_t) {
```
Comment (jsquyres) [inline opal/mca/pmix/pmix112/pmix/src/usock/usock_sendrecv.c]: Do you need a lock around accessing the members of this list?  I.e., with a name like "globals", does that imply that another thread may be accessing it at the same time?

Comment (rhc54) [inline opal/mca/pmix/pmix112/pmix/src/usock/usock_sendrecv.c]: no, we are in an event and therefore safe from any other thread access

File: opal/mca/pmix/pmix112/pmix/src/usock/usock_sendrecv.c [hunk 2]
```diff
@@ -99,15 +101,22 @@ static void lost_connection(pmix_peer_t *peer, pmix_status_t err)
...
+            if (reginfoptr->peer == peer) {
+                PMIX_RELEASE(reginfoptr);
```
Comment (jsquyres) [inline opal/mca/pmix/pmix112/pmix/src/usock/usock_sendrecv.c]: Per IM discussion -- @rhc54 is going to bring over another change here for removing the item from the list.

**Comment by hppritcha** [comment]
Travis looks kind of borked up again.  I'll give it another 30 minutes

**Pull request merged**